content/departments/cloud/technical-docs/private-registry/index.md

```diff
@@ -0,0 +1,84 @@
+
+Registries access is required for code intel auto-indexing to be working.
+
+## Public Registry with private domain
+
+Customer might expose their private registry via external loadbalancer. In this case registry will be accessible via public IP and Sourcegraph will provide customer 2 static NAT IPs for IP allowlist.
+If customer uses different (private) domain inside their repositories configuration for registry, additional change has to be performed in Cloud executors to translate customer private registry domain -> customer public loadbalancer IP.
+This is done via dns-proxy running on executors VM.
+
+Configuration:
+
+1. Modify config.yaml with additional entry:
+
+```yaml
+spec:
+  infrastructure:
+    executorPrivateDNS:
+      proxyTag: 202311201628
+      routes:
+        - destination: customer.loadbalancer.public.domain
+          source: customer.registry.private.domain
+```
+
+> Note: as customer loadbalancer is handling connections also for private domain (`customer.registry.private.domain`), it has to be configured as NLB TLS passthrough (certificate is validated on registry) or has to have both certificates for public and private domain installed.
+
+2. Open PR
+3. When approved and merged, executors can access customer public loadbalancer using customer private domain.
+
+## AWS Private Registry
+
+If customer uses private registry deployed in AWS VPC, customer can expose this registry via AWS Private Link.
+Sourcegraph will connect Cloud instance Google VPC with customer dedicated AWS VPC, where VPC Endpoint for customer VPC Endpoint Service will be created.
+
+1. [Create GCP<->AWS VPN connection](../private-code-hosts/index.md#creation-process) documentation
+2. If customer uses private domain for the reguistry exposed via VPC Endpoint Service, executors require dns-proxy to resolve private domain to VPC Endpoint.
+3. For VPC Endpoint created in step 1, obtain endpoint dns entry via:
+
+```sh
+cd terrasform/stacks/awsvpn
+for vpce in $(terraform state list | grep aws_vpc_endpoint | grep -v data); do terraform state show $vpce; done
+```
+
+- Modify config.yaml with additional entry:
+
+```yaml
+spec:
+  infrastructure:
+    executorPrivateDNS:
+      proxyTag: 202311201628
+      routes:
+        - source: private.CUSTOMER.DOMAIN
+          # VPC endpoint DNS entry - use domain with region inside the URL, b/c it routes to every zone inside VPC
+          destination: vpce-<ID>.vpce-svc-<HASH>.<REGION>.vpce.amazonaws.com
+```
+
+- Open PR
+- When approved and merged, executors can access customer public loadbalancer using customer private domain.
+
+## GCP Private Registry
+
+If customer uses private registry deployed in Google Cloud VPC, customer can expose the registry via Private Service Connect.
+
+1. [Create Private Service Connect](../private-code-hosts/index.md#gcp-private-service-connect-gcp-code-hosts-only) documentation.
```

Comment:
@michaellzc can we have any scenario when dns-proxy is required for PSC? When customer uses PSC, it makes it available via domain, and this domain is resolvable via [this domain](https://sourcegraph.sourcegraph.com/github.com/sourcegraph/controller@f44b5fa48a663c3f6615a7304d5c7b67d86aec1f/-/blob/internal/resource/psc/psc.go?L107) in GCP - executors can resolve it, so we only open IPTABLES in Fireckracker for this domain private IP.